### PR TITLE
Normalize usage to prefix_map

### PR DESCRIPTION
**Pull request by cthoyt**
All instances of `curie_map` and `prefixmap` have been normalized to `prefix_map` both in variable names and function names. The notable exception is when interacting with dictionaries coming from SSSOM yaml, where the name is unfortunately `curie_map` - so this is maintained as a constant string called `sssom.util.PREFIX_MAP_KEY` in case it's possible we can ever change it.

### CODE REVIEW

**Review by matentzn — APPROVED** [review]
You give me hope that a better open-source and open data world is possible. THANK YOU!

**Comment by cthoyt** [comment]
@matentzn can you adjust the settings so it's not possible to merge to the master branch without CI tests passing?

**Comment by matentzn** [comment]
I made it harder, but admin can still overwrite if necessary.. 

image

File: sssom/io.py
Comment (cthoyt) [reply sssom/io.py:37]: True, but you also can consider the context that the variable is being used, which is in a function called "parse"

Comment (matentzn) [reply sssom/io.py:37]: yes, but the variable does not refer to the "parse mode" - it refers to the way the prefix map should be assembled prior to parsing..

Comment (cthoyt) [reply sssom/io.py]: Okay you convinced me. I changed it in f6dbfb4

**Review by matentzn — APPROVED** [review]

**Comment by matentzn** [comment]
Thank you!

Somehow the merge button is not red after the approval is given.. but QC is not yet run. Annoying!

**Pull request merged**
